Automattic/jetpack
# Contact Form Block: Improve fallback markup.

This PR improves the Contact Form block markup rendered on non-frontend contexts. Rather than the `contact-form` shortcode literal or empty string that is currently displayed (depends on the context), let's give the user an actionable link to the page with the working contact form block.

This approach intentionally breaks our guideline of using `save` as the proper fallback given that this block requires child blocks to work so needs to be handled differently. See https://github.com/Automattic/jetpack/pull/17706#issuecomment-725438619.

Context | Before | After
--- | --- | ---
Email | <img width="633" alt="Screen Shot 2020-11-11 at 15 31 10" src="https://user-images.githubusercontent.com/1233880/98823959-fad28f00-2432-11eb-9bde-c25ca16d2378.png"> | <img width="626" alt="Screen Shot 2020-11-11 at 15 32 20" src="https://user-images.githubusercontent.com/1233880/98824060-1ccc1180-2433-11eb-880f-918d7782e828.png">
Notifications | <img width="400" alt="Screen Shot 2020-11-11 at 15 34 32" src="https://user-images.githubusercontent.com/1233880/98824289-6d436f00-2433-11eb-83b5-3d2ba74ab6bf.png"> | <img width="402" alt="Screen Shot 2020-11-11 at 15 35 13" src="https://user-images.githubusercontent.com/1233880/98824365-84825c80-2433-11eb-9bf8-0b99b5319904.png">
Reader | <img width="758" alt="Screen Shot 2020-11-11 at 15 35 47" src="https://user-images.githubusercontent.com/1233880/98824418-96fc9600-2433-11eb-800e-b2c31b90003e.png"> | <img width="763" alt="Screen Shot 2020-11-11 at 15 36 13" src="https://user-images.githubusercontent.com/1233880/98824482-a7147580-2433-11eb-8f9a-3e64df45de18.png">

#### Jetpack product discussion
* PCYsg-tvN-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

_Web_
- [Spin up a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=update/contact-form-block-save).
- Set up Jetpack and choose any plan (the free plan is enough).
- Go to WP Admin > Posts > Add new.
- Insert a "Contact Form" block.
- Publish the post.
- Visit the frontend of the post you published before.
- Make sure there form is rendered normally.

_API_
- [Go to the WP.com API console](https://developer.wordpress.com/docs/api/console/).
- Make a `GET` request to `/sites/:site/posts/:post-id` (where `:post-id` is the ID of the post you published above).
- Make sure the post content uses the fallback markup.
<img width="400" alt="Screen Shot 2020-11-11 at 15 19 44" src="https://user-images.githubusercontent.com/1233880/98823385-4df81200-2432-11eb-850d-edda6452a938.png">

_Email, notifications, and reader (only testable in WP.com)_
- Apply D52249-code to your WP.com sandbox.
- Point `public-api.wordpress.com` to the IP of your WP.com sandbox in your `/etc/hosts/` file.
- Go to https://wordpress.com/following/manage.
- Make sure to follow a test site you own.
- Enable the "Notify me of new posts" option.
<img width="200" alt="Screen Shot 2020-09-28 at 16 25 18" src="https://user-images.githubusercontent.com/1233880/94444997-3b3dbc80-01a7-11eb-850f-47e87d259098.png">

- Go to https://wordpress.com/post.
- Select the site you followed above.
- Publish a post that contains a "Contact Form" block.
- Grab the IDs of the post and the site where the post has been published.
- In a WP.com sandbox run `php ./bin/subscriptions/send.php <BLOG_ID> post <POST_ID> <YOUR_EMAIL>`.
- Check the email received and make sure it contains the fallback link.
<img width="400" alt="Screen Shot 2020-11-11 at 15 32 20" src="https://user-images.githubusercontent.com/1233880/98824060-1ccc1180-2433-11eb-880f-918d7782e828.png">

- Open the masterbar notifications.
- Make sure it includes the fallback link.
<img width="350" alt="Screen Shot 2020-11-11 at 15 35 13" src="https://user-images.githubusercontent.com/1233880/98824365-84825c80-2433-11eb-9bf8-0b99b5319904.png">

- Go to Reader.
- Open the post you just published above.
- Make sure it includes the fallback link.
<img width="400" alt="Screen Shot 2020-11-11 at 15 35 47" src="https://user-images.githubusercontent.com/1233880/98824418-96fc9600-2433-11eb-800e-b2c31b90003e.png"> | <img width="763" alt="Screen Shot 2020-11-11 at 15 36 13" src="https://user-images.githubusercontent.com/1233880/98824482-a7147580-2433-11eb-8f9a-3e64df45de18.png">

#### Proposed changelog entry for your changes:
No changelog needed.
